patterns/1-initial/secure-innersource.md

```diff
@@ -0,0 +1,144 @@
+
+And given 'findability' is also important, adding the additional level that facilitates findability, but still provides a light-weight gate seems to be a good way out to steer projects out of closed source.
+
+As concrete example, which allows repositories to be discovered, even though there is no company-wide access, consider the following sharing levels:
+
+- **PUBLIC** - open source: findable and accessible for all SW developers in the world
+- **INTERNAL** - inner source: findable and accessible for all SW developers in the company
+- **RESTRICTED** - restricted source: findable for all SW developers in the company, low-friction process to gain full access.
+- **CLOSED** - closed source: only accessible to named individuals and not findable via the search engine for other SW developers in the company.
+
+Repositories with RESTRICTED sharing level are included in the [central catalog][innersource-portal] and/or fed into the central search mechanism, together with their `README.md` and some meta information. Their content, however, is not accessible without following a light-weight access request process. This can for example be contacting one of the code owners or some other request mechanism described in the `README.md` file.
+
+Notes:
+
+- This will likely have an impact on contributions due to the additional step to access the code.
+- It is feasible to use a tool that automatically adds the `README.md` and some other meta-data to an internal repository in GitHub, allowing the GitHub search feature to include this data in the search results.
+
+# Resulting Context
+
+*What is the situation after the problem has been solved? The original context is changed indirectly by way of the solution. Often this section can include discussion of the next possible Patterns/problems introduced. This section can be short in content - the solution may not introduce new problems or change much context.*
+
+Many times the InnerSource adoption in an organization will [start as an experiment][start-as-an-experiment], with a small number of SHARED repositories.
+
+If the task force (and with that the Security Team) collaborate in this early phase already, they can make joint experiences with how security-related issues can be identified and addressed. This also allows the Security Teams to build up more confidence on InnerSource from a security angle.
+
+With the increased confidence and lessons learned from this experiment, the task force faces less pushback when rolling out InnerSource to more repositories.
+
```

Comment:
Consider adding

The defined sharing levels for repositories provide guidance on what sharing level to use for what type of repository content. The increased security awareness of the engineers and more secure set up of the source code management systems increases overall software development security.

Reply:
Done.